app/services/data_sources/json_user_data_source.rb

```diff
@@ -180,5 +184,20 @@ def create_user!(id: nil, email: nil, password: nil, provider_credentials: [])
     def find_user(id: nil)
       return users.detect { |user| user.id == id }
     end
+
+    private
+
+    # Finds the index of the user, and returns an existing `user` if they exist
+    #
+    # @param  [User] `user` a user to lookup by `user.email`
+    # @return [Integer] `user_index` in the `users` array
+    # @return [User] `existing_user` in the `users.json` file
+    def find_user_index_and_existing_user(user:)
+      users.each.with_index do |old_user, index|
+        return [index, old_user] if old_user.email.casecmp(user.email.downcase).zero?
```

Comment:
maybe instead of email, we should compare ID? I know right now email is pretty much guaranteed to be unique, but ID is truly guaranteed to be unique

Reply:
Yeah, I think I wrote that code you copied 😆 My original thought was that the update might be operating on a `User` that didn't yet contain an ID. But really, I should make that much more clear if that's what I was doing.

Reply:
Ahh, gotcha! Thank you for clearing that up :)